internal/provider/network_connection_resource.go

```diff
@@ -0,0 +1,172 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/doublecloud/go-genproto/doublecloud/network/v1"
+	dcsdk "github.com/doublecloud/go-sdk"
+	dcgennet "github.com/doublecloud/go-sdk/gen/network"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func NewNetworkConnectionResource() resource.Resource {
+	return &NetworkConnectionResource{}
+}
+
+type NetworkConnectionResource struct {
+	sdk                      *dcsdk.SDK
+	networkConnectionService *dcgennet.NetworkConnectionServiceClient
+}
+
+func (r *NetworkConnectionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_network_connection"
+}
+
+func (r *NetworkConnectionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = networkConnectionResourceSchema
+}
+
+func (r *NetworkConnectionResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	sdk, ok := req.ProviderData.(*dcsdk.SDK)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *dcsdk.SDK, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.sdk = sdk
+	r.networkConnectionService = r.sdk.Network().NetworkConnection()
+}
+
+func (r *NetworkConnectionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *NetworkConnectionModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createReq := &network.CreateNetworkConnectionRequest{
+		NetworkId:   data.NetworkID.ValueString(),
+		Params:      nil,
+		Description: data.Description.ValueString(),
+	}
+
+	switch {
+	case data.AWS != nil:
+		aws := &network.CreateAWSNetworkConnectionRequest{}
+		createReq.Params = &network.CreateNetworkConnectionRequest_Aws{
+			Aws: aws,
+		}
+		switch {
+		case data.AWS.Peering != nil:
+			aws.Type = &network.CreateAWSNetworkConnectionRequest_Peering{
+				Peering: &network.CreateAWSNetworkConnectionPeeringRequest{
+					VpcId:         data.AWS.Peering.VPCID.ValueString(),
+					AccountId:     data.AWS.Peering.AccountID.ValueString(),
+					RegionId:      data.AWS.Peering.RegionID.ValueString(),
+					Ipv4CidrBlock: data.AWS.Peering.IPv4CIDRBlock.ValueString(),
+					Ipv6CidrBlock: data.AWS.Peering.IPv6CIDRBlock.ValueString(),
+				},
+			}
+		default:
+			resp.Diagnostics.AddError("misconfiguration", "\"aws.peering\" must be specified")
```

Comment:
Move it to schema validation and remove unnecessary switch.

Reply:
It is validated in schema. Here is just for
1. double check
2. TGW network connection support in future

Reply:
Well, let's modify this place on the way of TGW implementing.

Reply:
let's keep it as golang double check